src/univariate/affine.jl

```diff
@@ -0,0 +1,222 @@
+
+# Support
+function minimum(d::AffineDistribution)
+    minim = d.scale > 0 ? minimum : maximum
+    return d.shift + d.scale * minim(d.base_dist)
+end
+function maximum(d::AffineDistribution)
+    maxim = d.scale > 0 ? maximum : minimum
+    return d.shift + d.scale * maxim(d.base_dist)
+end
+support(d::AffineDistribution) = affine_support(d.shift, d.scale, support(d.base_dist))
+
+affine_support(shift::Real, scale::Real, support) = shift .+ scale .* support
+function affine_support(shift::Real, scale::Real, support::RealInterval)
+    if scale > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(shift + scale * lower, shift + scale * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{T}}, shift::Real, scale::Real, base_dist::UnivariateDistribution) where T<:Real
+    return AffineDistribution(T(shift), T(scale), base_dist)
+end
+function convert(::Type{AffineDistribution{T}}, d::AffineDistribution{S}) where {T<:Real, S<:Real} 
+    # Should we really leave the type of d.base_dist unchanged? (Old LocationScale behavior)
+    return affine(T(d.shift), T(d.scale), d.base_dist)
+end
+
+Base.eltype(::Type{<:AffineDistribution{T,S,D}}) where {T,S,D} = promote(T, eltype(D))
+
+
+#### Parameters
+
+function location(d::AffineDistribution)
+    if hasmethod(location, (typeof(d.base_dist),))
+        return d.shift + location(d.base_dist)
+    else
+        return d.shift
+    end
+end
+function scale(d::AffineDistribution)
+    if hasmethod(scale, (typeof(d.base_dist),))
+        return d.scale * scale(d.base_dist)
+    else
+        return d.scale
+    end
+end
+params(d::AffineDistribution) = (d.shift, d.scale, d.base_dist)
+partype(::AffineDistribution{T}) where {T} = T
+
+
+#### Statistics
+
+mean(d::AffineDistribution) = d.shift + d.scale * mean(d.base_dist)
+median(d::AffineDistribution) = d.shift + d.scale * median(d.base_dist)
+mode(d::AffineDistribution) = d.shift + d.scale * mode(d.base_dist)
+modes(d::AffineDistribution) = d.shift .+ d.scale .* modes(d.base_dist)
+
+var(d::AffineDistribution) = d.scale^2 * var(d.base_dist)
+std(d::AffineDistribution) = abs(d.scale) * std(d.base_dist)
+skewness(d::AffineDistribution) = skewness(d.base_dist)
+kurtosis(d::AffineDistribution) = kurtosis(d.base_dist)
+
+isplatykurtic(d::AffineDistribution) = isplatykurtic(d.base_dist)
+isleptokurtic(d::AffineDistribution) = isleptokurtic(d.base_dist)
+ismesokurtic(d::AffineDistribution) = ismesokurtic(d.base_dist)
+
+entropy(d::ContinuousAffine) = entropy(d.base_dist) + log(abs(d.scale))
+entropy(d::DiscreteAffine) = entropy(d.base_dist)
+
+mgf(d::AffineDistribution, t::Real) = exp(d.shift * t) * mgf(d.base_dist, d.scale * t)
+
+
+#### Evaluation & Sampling
+
+pdf(d::ContinuousAffine, x::Real) = pdf(d.base_dist, (x - d.shift) / d.scale) / abs(d.scale)
+pdf(d::DiscreteAffine, x::Real) = pdf(d.base_dist, (x - d.shift) / d.scale)
```

Comment:
Let's think if this is right for integer valued distributions:
```
julia> x = 1<<60 + 1
1152921504606846977

julia> (x*5)/5 == x
false

julia> (x*5)/5 ≈ x
true
```

Reply:
For integer-valued distributions one has to work with `Rational`s.

Reply:
As done eg in the tests.

Reply:
Ah, nice, let's add a remark to the docs then